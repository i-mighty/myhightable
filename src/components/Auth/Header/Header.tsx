import React from 'react';
import styled from 'styled-components/native';
import { Header as NBHeader, Text, Left, Body } from 'native-base';
import colorOpacity from '@src/utils/color-opacity';
import Banner from '@src/assets/highTableLogo.svg';

interface HeaderProps {
  testID?: string;
}

const StyledHeader = styled(NBHeader)<HeaderProps>`
  background-color: ${({ theme }) => colorOpacity(theme.vars.dark, 0.5)};
  width: 100%;
`;

const ButtonText = styled(Text)`
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  text-align: center;
  color: ${({ theme }) => colorOpacity(theme.vars.white, 0.5)};
`;

const Header: React.FC<HeaderProps> = ({ testID }) => {
  return (
    <StyledHeader testID={testID} transparent noLeft>
      <Body>
        <Banner width={149} height={38} testID="banner" />
      </Body>
    </StyledHeader>
  );
};

export default Header;
