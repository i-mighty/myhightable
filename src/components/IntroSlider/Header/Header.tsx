import React from 'react';
import styled from 'styled-components/native';
import { Header as NBHeader, Button, Right, Text } from 'native-base';
import colorOpacity from '@src/utils/color-opacity';

interface HeaderProps {
  buttonText?: string;
  onButtonClick?: () => void;
  testID?: string;
  buttonTestID?: string;
}

const StyledHeader = styled(NBHeader)<HeaderProps>`
  background-color: ${({ theme }) => colorOpacity(theme.vars.dark, 0.5)};
  width: 100%;
`;

const ButtonText = styled(Text)`
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  text-align: center;
  color: ${({ theme }) => colorOpacity(theme.vars.white, 0.5)};
`;

const Header: React.FC<HeaderProps> = ({
  buttonText,
  onButtonClick,
  testID,
  buttonTestID,
}) => {
  return buttonText ? (
    <StyledHeader testID={testID} transparent>
      <Right>
        <Button testID={buttonTestID} transparent onPress={onButtonClick}>
          <ButtonText>{buttonText}</ButtonText>
        </Button>
      </Right>
    </StyledHeader>
  ) : (
    <StyledHeader testID={testID} transparent />
  );
};

export default Header;
