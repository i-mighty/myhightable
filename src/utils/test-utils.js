import { ThemeProvider } from 'styled-components/native';
import React from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (Component, props) => {
  baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
