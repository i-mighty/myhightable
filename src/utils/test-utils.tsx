import { ThemeProvider } from 'styled-components/native';
import React, { ReactElement } from 'react';
import { render as baseRender } from '@testing-library/react-native';
import theme from '@src/stores/theme';

export const renderWithProps = (
  Component: React.ElementType,
  props: Object,
) => {
  return baseRender(
    <ThemeProvider theme={theme}>
      <Component {...props} />
    </ThemeProvider>,
  );
};
