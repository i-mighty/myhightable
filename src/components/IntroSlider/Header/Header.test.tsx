import Header from './Header';
import { renderWithProps } from '@src/utils/test-utils';
import { fireEvent, NativeTestEvent } from '@testing-library/react-native';

const onButtonClick = jest.fn(() => {
  console.log('Pressed');
});

const defaultProps = { testID: 'header', buttonTestID: 'button' };
const buttonTextProps = { ...defaultProps, buttonText: 'Skip' };
const buttonClickProps = { ...buttonTextProps, onButtonClick };

describe('Header', () => {
  it('renders properly without button', () => {
    const { getByTestId, baseElement, queryByTestId } = renderWithProps(
      Header,
      defaultProps,
    );
    expect(getByTestId(defaultProps.testID)).toBeDefined();
    expect(queryByTestId(defaultProps.buttonTestID)).toBeNull();
    expect(baseElement).toMatchSnapshot();
  });

  it('renders properly with button', () => {
    const { getByTestId, getByText, baseElement } = renderWithProps(
      Header,
      buttonTextProps,
    );
    const header = getByTestId(buttonTextProps.testID);
    const button = getByTestId(buttonTextProps.buttonTestID);
    const buttonText = getByText(buttonTextProps.buttonText);
    expect(header).toBeDefined();
    expect(button).toBeDefined();
    expect(buttonText).toBeDefined();
    expect(buttonText.props.children).toBe(buttonTextProps.buttonText);
    expect(baseElement).toMatchSnapshot();
  });

  it('handles button click event', async () => {
    const { getByTestId, getByText, baseElement } = renderWithProps(
      Header,
      buttonClickProps,
    );
    const header = getByTestId(buttonClickProps.testID);
    const button = getByTestId(buttonClickProps.buttonTestID);
    const buttonText = getByText(buttonClickProps.buttonText);
    expect(header).toBeDefined();
    expect(button).toBeDefined();
    expect(buttonText).toBeDefined();
    expect(buttonText.props.children).toBe(buttonClickProps.buttonText);
    fireEvent(button, new NativeTestEvent('press'));
    expect(buttonClickProps.onButtonClick).toHaveBeenCalled();
    expect(baseElement).toMatchSnapshot();
  });
});
