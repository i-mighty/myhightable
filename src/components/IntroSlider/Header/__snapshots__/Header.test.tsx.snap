// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header handles button click event 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <SafeAreaView
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "transparent",
                "borderBottomWidth": 0.5,
                "elevation": 0,
                "flexDirection": "row",
                "height": 64,
                "justifyContent": "center",
                "left": 0,
                "paddingLeft": 6,
                "paddingRight": 10,
                "paddingTop": 18,
                "right": 0,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
                "top": 0,
              },
              Object {
                "backgroundColor": "rgba(0,0,0,0.5)",
                "width": "100%",
              },
            ]
          }
          testID="header"
          transparent={true}
        >
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "alignSelf": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <
              activeOpacity={0.5}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": null,
                  "backgroundColor": "transparent",
                  "borderBottomWidth": null,
                  "borderColor": null,
                  "borderLeftWidth": null,
                  "borderRadius": 50,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 0,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "space-between",
                  "marginRight": -9,
                  "paddingBottom": 6,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 6,
                  "shadowColor": null,
                  "shadowOffset": null,
                  "shadowOpacity": null,
                  "shadowRadius": null,
                }
              }
              testID="button"
              transparent={true}
            >
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#007aff",
                      "fontFamily": "System",
                      "fontSize": 17,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "paddingLeft": 16,
                      "paddingRight": 0,
                      "top": 1,
                    },
                    Object {
                      "color": "rgba(255,255,255,0.5)",
                      "fontFamily": "Roboto",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "lineHeight": 16,
                      "textAlign": "center",
                    },
                  ]
                }
                uppercase={false}
              >
                Skip
              </Text>
            </>
          </View>
        </View>
      </SafeAreaView>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": "100%",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingBottom": 4,
          "paddingEnd": 4,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <TouchableWithoutFeedback
        hitSlop={
          Object {
            "bottom": 4,
            "left": 4,
            "right": 4,
            "top": 4,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(250, 186, 48, 0.95)",
              },
              Object {
                "borderRadius": 14,
                "height": 28,
                "justifyContent": "center",
                "paddingHorizontal": 12,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 12,
                "includeFontPadding": false,
                "lineHeight": 16,
              }
            }
          >
            Dismiss All
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
    <ScrollView
      data={
        Array [
          Object {
            "category": "Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?﻿%s﻿%s",
            "warnings": Array [
              YellowBoxWarning {
                "message": Object {
                  "content": "Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of \`Button\`.
    in TouchableOpacity (at Button.js:79)
    in Button (created by Styled(Button))
    in Styled(Button) (created by Header)
    in View (created by View)
    in View (at Right.js:10)
    in Right (created by Styled(Right))
    in Styled(Right) (created by Header)
    in View (created by View)
    in View (at Header.js:54)
    in SafeAreaView (created by SafeAreaView)
    in SafeAreaView (at Header.js:49)
    in View (created by View)
    in View (at Header.js:33)
    in Header (created by Styled(Header))
    in Styled(Header) (created by Context.Consumer)
    in StyledNativeComponent (created by Styled(Styled(Header)))
    in Styled(Styled(Header)) (created by Header)
    in Header
    in ThemeProvider
    in View (created by View)
    in View (created by AppContainer)
    in View (created by View)
    in View (created by AppContainer)
    in AppContainer (at src/index.js:26)",
                  "substitutions": Array [
                    Object {
                      "length": 38,
                      "offset": 120,
                    },
                    Object {
                      "length": 891,
                      "offset": 158,
                    },
                  ],
                },
                "stack": Array [
                  Object {
                    "arguments": Array [],
                    "column": 29,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-native\\\\Libraries\\\\YellowBox\\\\Data\\\\YellowBoxWarning.js",
                    "lineNumber": 58,
                    "methodName": "Function.parse",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 56,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-native\\\\Libraries\\\\YellowBox\\\\YellowBox.js",
                    "lineNumber": 180,
                    "methodName": "registerError",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 21,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-native\\\\Libraries\\\\YellowBox\\\\YellowBox.js",
                    "lineNumber": 84,
                    "methodName": "errorImpl",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 13,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-native\\\\Libraries\\\\YellowBox\\\\YellowBox.js",
                    "lineNumber": 63,
                    "methodName": "BufferedConsole.Object.<anonymous>.console.error",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 29,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-test-renderer\\\\cjs\\\\react-test-renderer.development.js",
                    "lineNumber": 120,
                    "methodName": "printWarning",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 4,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-test-renderer\\\\cjs\\\\react-test-renderer.development.js",
                    "lineNumber": 92,
                    "methodName": "error",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 8,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-test-renderer\\\\cjs\\\\react-test-renderer.development.js",
                    "lineNumber": 8004,
                    "methodName": "validateFunctionComponentInDev",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 6,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-test-renderer\\\\cjs\\\\react-test-renderer.development.js",
                    "lineNumber": 7971,
                    "methodName": "mountIndeterminateComponent",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 15,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-test-renderer\\\\cjs\\\\react-test-renderer.development.js",
                    "lineNumber": 9019,
                    "methodName": "beginWork",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 11,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-test-renderer\\\\cjs\\\\react-test-renderer.development.js",
                    "lineNumber": 12649,
                    "methodName": "performUnitOfWork",
                  },
                ],
                "symbolicated": Object {
                  "error": null,
                  "stack": null,
                  "status": "NONE",
                },
              },
            ],
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      maxToRenderPerBatch={10}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      scrollEnabled={false}
      scrollEventThrottle={50}
      scrollsToTop={false}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "height": 48.5,
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View
        style={null}
      >
        <TouchableWithoutFeedback>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(250, 186, 48, 0.95)",
                },
                Object {
                  "height": 48,
                  "justifyContent": "center",
                  "marginTop": 0.5,
                  "paddingHorizontal": 12,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 1)",
                    "flex": 1,
                    "fontSize": 14,
                    "includeFontPadding": false,
                    "lineHeight": 18,
                  }
                }
              >
                <Text>
                  Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
                </Text>
                <Text
                  style={
                    Object {
                      "color": "rgba(255, 255, 255, 0.6)",
                    }
                  }
                >
                  

Check the render method of \`Button\`.
                </Text>
                <Text
                  style={
                    Object {
                      "color": "rgba(255, 255, 255, 0.6)",
                    }
                  }
                >
                  
    in TouchableOpacity (at Button.js:79)
    in Button (created by Styled(Button))
    in Styled(Button) (created by Header)
    in View (created by View)
    in View (at Right.js:10)
    in Right (created by Styled(Right))
    in Styled(Right) (created by Header)
    in View (created by View)
    in View (at Header.js:54)
    in SafeAreaView (created by SafeAreaView)
    in SafeAreaView (at Header.js:49)
    in View (created by View)
    in View (at Header.js:33)
    in Header (created by Styled(Header))
    in Styled(Header) (created by Context.Consumer)
    in StyledNativeComponent (created by Styled(Styled(Header)))
    in Styled(Styled(Header)) (created by Header)
    in Header
    in ThemeProvider
    in View (created by View)
    in View (created by AppContainer)
    in View (created by View)
    in View (created by AppContainer)
    in AppContainer (at src/index.js:26)
                </Text>
              </Text>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </ScrollView>
    <SafeAreaView
      style={
        Object {
          "backgroundColor": "rgba(250, 186, 48, 0.95)",
          "marginTop": 0.5,
        }
      }
    />
  </View>
</View>
`;

exports[`Header renders properly with button 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <SafeAreaView
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "transparent",
                "borderBottomWidth": 0.5,
                "elevation": 0,
                "flexDirection": "row",
                "height": 64,
                "justifyContent": "center",
                "left": 0,
                "paddingLeft": 6,
                "paddingRight": 10,
                "paddingTop": 18,
                "right": 0,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
                "top": 0,
              },
              Object {
                "backgroundColor": "rgba(0,0,0,0.5)",
                "width": "100%",
              },
            ]
          }
          testID="header"
          transparent={true}
        >
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "alignSelf": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <
              activeOpacity={0.5}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": null,
                  "backgroundColor": "transparent",
                  "borderBottomWidth": null,
                  "borderColor": null,
                  "borderLeftWidth": null,
                  "borderRadius": 50,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 0,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "space-between",
                  "marginRight": -9,
                  "paddingBottom": 6,
                  "paddingLeft": 15,
                  "paddingRight": 15,
                  "paddingTop": 6,
                  "shadowColor": null,
                  "shadowOffset": null,
                  "shadowOpacity": null,
                  "shadowRadius": null,
                }
              }
              testID="button"
              transparent={true}
            >
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#007aff",
                      "fontFamily": "System",
                      "fontSize": 17,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "paddingLeft": 16,
                      "paddingRight": 0,
                      "top": 1,
                    },
                    Object {
                      "color": "rgba(255,255,255,0.5)",
                      "fontFamily": "Roboto",
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                      "lineHeight": 16,
                      "textAlign": "center",
                    },
                  ]
                }
                uppercase={false}
              >
                Skip
              </Text>
            </>
          </View>
        </View>
      </SafeAreaView>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "bottom": "100%",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "paddingBottom": 4,
          "paddingEnd": 4,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <TouchableWithoutFeedback
        hitSlop={
          Object {
            "bottom": 4,
            "left": 4,
            "right": 4,
            "top": 4,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(250, 186, 48, 0.95)",
              },
              Object {
                "borderRadius": 14,
                "height": 28,
                "justifyContent": "center",
                "paddingHorizontal": 12,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 12,
                "includeFontPadding": false,
                "lineHeight": 16,
              }
            }
          >
            Dismiss All
          </Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
    <ScrollView
      data={
        Array [
          Object {
            "category": "Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?﻿%s﻿%s",
            "warnings": Array [
              YellowBoxWarning {
                "message": Object {
                  "content": "Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of \`Button\`.
    in TouchableOpacity (at Button.js:79)
    in Button (created by Styled(Button))
    in Styled(Button) (created by Header)
    in View (created by View)
    in View (at Right.js:10)
    in Right (created by Styled(Right))
    in Styled(Right) (created by Header)
    in View (created by View)
    in View (at Header.js:54)
    in SafeAreaView (created by SafeAreaView)
    in SafeAreaView (at Header.js:49)
    in View (created by View)
    in View (at Header.js:33)
    in Header (created by Styled(Header))
    in Styled(Header) (created by Context.Consumer)
    in StyledNativeComponent (created by Styled(Styled(Header)))
    in Styled(Styled(Header)) (created by Header)
    in Header
    in ThemeProvider
    in View (created by View)
    in View (created by AppContainer)
    in View (created by View)
    in View (created by AppContainer)
    in AppContainer (at src/index.js:26)",
                  "substitutions": Array [
                    Object {
                      "length": 38,
                      "offset": 120,
                    },
                    Object {
                      "length": 891,
                      "offset": 158,
                    },
                  ],
                },
                "stack": Array [
                  Object {
                    "arguments": Array [],
                    "column": 29,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-native\\\\Libraries\\\\YellowBox\\\\Data\\\\YellowBoxWarning.js",
                    "lineNumber": 58,
                    "methodName": "Function.parse",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 56,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-native\\\\Libraries\\\\YellowBox\\\\YellowBox.js",
                    "lineNumber": 180,
                    "methodName": "registerError",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 21,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-native\\\\Libraries\\\\YellowBox\\\\YellowBox.js",
                    "lineNumber": 84,
                    "methodName": "errorImpl",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 13,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-native\\\\Libraries\\\\YellowBox\\\\YellowBox.js",
                    "lineNumber": 63,
                    "methodName": "BufferedConsole.Object.<anonymous>.console.error",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 29,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-test-renderer\\\\cjs\\\\react-test-renderer.development.js",
                    "lineNumber": 120,
                    "methodName": "printWarning",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 4,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-test-renderer\\\\cjs\\\\react-test-renderer.development.js",
                    "lineNumber": 92,
                    "methodName": "error",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 8,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-test-renderer\\\\cjs\\\\react-test-renderer.development.js",
                    "lineNumber": 8004,
                    "methodName": "validateFunctionComponentInDev",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 6,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-test-renderer\\\\cjs\\\\react-test-renderer.development.js",
                    "lineNumber": 7971,
                    "methodName": "mountIndeterminateComponent",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 15,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-test-renderer\\\\cjs\\\\react-test-renderer.development.js",
                    "lineNumber": 9019,
                    "methodName": "beginWork",
                  },
                  Object {
                    "arguments": Array [],
                    "column": 11,
                    "file": "C:\\\\Users\\\\Josiah\\\\Documents\\\\AppProjects\\\\hightable\\\\node_modules\\\\react-test-renderer\\\\cjs\\\\react-test-renderer.development.js",
                    "lineNumber": 12649,
                    "methodName": "performUnitOfWork",
                  },
                ],
                "symbolicated": Object {
                  "error": null,
                  "stack": null,
                  "status": "NONE",
                },
              },
            ],
          },
        ]
      }
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      maxToRenderPerBatch={10}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      scrollEnabled={false}
      scrollEventThrottle={50}
      scrollsToTop={false}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "height": 48.5,
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View
        style={null}
      >
        <TouchableWithoutFeedback>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(250, 186, 48, 0.95)",
                },
                Object {
                  "height": 48,
                  "justifyContent": "center",
                  "marginTop": 0.5,
                  "paddingHorizontal": 12,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                numberOfLines={2}
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 1)",
                    "flex": 1,
                    "fontSize": 14,
                    "includeFontPadding": false,
                    "lineHeight": 18,
                  }
                }
              >
                <Text>
                  Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?
                </Text>
                <Text
                  style={
                    Object {
                      "color": "rgba(255, 255, 255, 0.6)",
                    }
                  }
                >
                  

Check the render method of \`Button\`.
                </Text>
                <Text
                  style={
                    Object {
                      "color": "rgba(255, 255, 255, 0.6)",
                    }
                  }
                >
                  
    in TouchableOpacity (at Button.js:79)
    in Button (created by Styled(Button))
    in Styled(Button) (created by Header)
    in View (created by View)
    in View (at Right.js:10)
    in Right (created by Styled(Right))
    in Styled(Right) (created by Header)
    in View (created by View)
    in View (at Header.js:54)
    in SafeAreaView (created by SafeAreaView)
    in SafeAreaView (at Header.js:49)
    in View (created by View)
    in View (at Header.js:33)
    in Header (created by Styled(Header))
    in Styled(Header) (created by Context.Consumer)
    in StyledNativeComponent (created by Styled(Styled(Header)))
    in Styled(Styled(Header)) (created by Header)
    in Header
    in ThemeProvider
    in View (created by View)
    in View (created by AppContainer)
    in View (created by View)
    in View (created by AppContainer)
    in AppContainer (at src/index.js:26)
                </Text>
              </Text>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </ScrollView>
    <SafeAreaView
      style={
        Object {
          "backgroundColor": "rgba(250, 186, 48, 0.95)",
          "marginTop": 0.5,
        }
      }
    />
  </View>
</View>
`;

exports[`Header renders properly without button 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <SafeAreaView
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.5)",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "transparent",
                "borderBottomWidth": 0.5,
                "elevation": 0,
                "flexDirection": "row",
                "height": 64,
                "justifyContent": "center",
                "left": 0,
                "paddingLeft": 6,
                "paddingRight": 10,
                "paddingTop": 18,
                "right": 0,
                "shadowColor": null,
                "shadowOffset": null,
                "shadowOpacity": null,
                "shadowRadius": null,
                "top": 0,
              },
              Object {
                "backgroundColor": "rgba(0,0,0,0.5)",
                "width": "100%",
              },
            ]
          }
          testID="header"
          transparent={true}
        />
      </SafeAreaView>
    </View>
  </View>
</View>
`;
